### PR TITLE
Add generic methods for get and post

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: erdjs/src/proxyProvider.ts [hunk 1]
```diff
@@ -16,155 +16,171 @@ const JSONbig = require("json-bigint");
...
+      return payload;
+    } catch (error) {
+      if (!error.response) {
+        Logger.warn(error);
+        throw new errors.ErrApiProviderGet(resourceUrl, error.toString(), error);
+      }
+
+      let errorData = error.response.data;
+      let originalErrorMessage = errorData.error || errorData.message || JSON.stringify(errorData);
+      throw new errors.ErrApiProviderGet(resourceUrl, originalErrorMessage, error);
```
Comment (camilbancioiu) [inline erdjs/src/proxyProvider.ts:141]: This is similar to `ApiProvider.handleApiError()`.

The comment at the top of the file says that we might merge `ProxyProvider` into `ApiProvider`. Maybe creating the function `ProxyProvider.handleApiError()`, just like in `ApiProvider`, will help with this merge.

File: erdjs/src/proxyProvider.ts [hunk 2]
```diff
@@ -16,155 +16,171 @@ const JSONbig = require("json-bigint");
...
+      return responsePayload;
+    } catch (error) {
+      if (!error.response) {
+        Logger.warn(error);
+        throw new errors.ErrApiProviderPost(resourceUrl, error.toString(), error);
+      }
+
+      let errorData = error.response.data;
+      let originalErrorMessage = errorData.error || errorData.message || JSON.stringify(errorData);
+      throw new errors.ErrApiProviderPost(resourceUrl, originalErrorMessage, error);
```
Comment (camilbancioiu) [inline erdjs/src/proxyProvider.ts:164]: Optionally use `handleApiError()`.

**Review by camilbancioiu — DISMISSED** [review]

File: erdjs/src/proxyProvider.ts [hunk 3]
```diff
@@ -16,155 +16,171 @@ const JSONbig = require("json-bigint");
...
+  async doPostGeneric(resourceUrl: string, payload: any, callback: (response: any) => any): Promise<any> {
+    let response = await this.doPost(resourceUrl, payload);
+    return callback(response);
+  }
+
+  private async doGet(resourceUrl: string): Promise<any> {
+    try {
+      let url = `${this.url}/${resourceUrl}`;
+      let response = await axios.get(url, { timeout: this.timeoutLimit });
+      let payload = response.data.data;
```
Comment (bogdan-rosianu) [inline erdjs/src/proxyProvider.ts:131]: maybe check for status code != 200 ?

File: erdjs/src/proxyProvider.ts [hunk 4]
```diff
@@ -16,155 +16,171 @@ const JSONbig = require("json-bigint");
...
+  private async doPost(resourceUrl: string, payload: any): Promise<any> {
+    try {
+      let url = `${this.url}/${resourceUrl}`;
+      let response = await axios.post(url, payload, {
+        timeout: this.timeoutLimit,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      let responsePayload = response.data.data;
```
Comment (bogdan-rosianu) [inline erdjs/src/proxyProvider.ts:154]: also here. maybe check for code != 200 ?

**Review by bogdan-rosianu — DISMISSED** [review]

File: erdjs/src/proxyProvider.ts (hunk 1 above)
Comment (danielailie) [reply erdjs/src/proxyProvider.ts]: done

File: erdjs/src/proxyProvider.ts (hunk 2 above)
Comment (danielailie) [reply erdjs/src/proxyProvider.ts]: done

**Review by camilbancioiu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**